source/common/config/grpc_stream.h

```diff
@@ -109,11 +96,33 @@ class GrpcStream : public Grpc::TypedAsyncStreamCallbacks<ResponseProto>,
 private:
   void drainRequests() {
     ENVOY_LOG(trace, "draining discovery requests {}", request_queue_.size());
-    while (!request_queue_.empty() && checkRateLimitAllowsDrain(request_queue_.size())) {
+    while (!request_queue_.empty() && checkRateLimitAllowsDrain()) {
       // Process the request, if rate limiting is not enabled at all or if it is under rate limit.
       sendDiscoveryRequest(request_queue_.front());
       request_queue_.pop();
     }
+    // Although request_queue_.push() happens elsewhere, the only time the queue is non-transiently
+    // non-empty is when it remains non-empty after a drain attempt. (The push() doesn't matter
+    // because we always attempt this drain immediately after the push). Basically, a change in
+    // queue length is not "meaningful" until it has persisted until here. Since we are constantly
+    // doing set(0), we need the if(>0 || used) to keep this stat from being wrongly marked
+    // interesting and needlessly taking up space. The first time we set(123), used becomes true,
```

Comment:
I don't see how the `set()` makes `used` true below... we can chat IRL tomorrow if that's easier.

Reply:
Stats have used() false until they are first given a value; for a gauge that includes set(), even set(0).  https://github.com/envoyproxy/envoy/blob/master/include/envoy/stats/stats.h#L58    and implementation at   https://github.com/envoyproxy/envoy/blob/master/source/common/stats/stat_data_allocator_impl.h#L134